File: Lab 6/src/main/java/SalesManager.java
import java.util.ArrayList;
import java.util.Comparator;
import java.util.LinkedList;
import java.util.List;

public class SalesManager {
    private List<Accommodation> accommodationList = new ArrayList<>();

    public List<Accommodation> findByPrice(List<Accommodation> list, double price) {
        List<Accommodation> result = new LinkedList<>();
        for (Accommodation i : list) {
            if (i.getPrice() < price) {
                result.add(i);
            }
        }
        return result;
    }

    public List<Accommodation> findByDistance(List<Accommodation> list, ListLocationOfSIO listLocationOfSIO,
                                              double distance) {
        List<Accommodation> result = new LinkedList<>();
        for (Accommodation accomodation : list) {
            if ((accomodation.getLocation().getLatitude() - listLocationOfSIO.getLatitude()) < distance
                    || (listLocationOfSIO.getLatitude() - accomodation.getLocation().getLatitude()) < distance) {
                result.add(accomodation);
            }
        }
        return result;
    }

    public void sortAccomodationByPrice(List<Accommodation> accommodations) {
        accommodations.sort(Comparator.comparing(Accommodation::getPrice));
    }

    public List<Accommodation> getAccommodationList() {
        return accommodationList;
    }
}

File: Lab 13 (lab 4 python)/app/sales_manager.py
from operator import attrgetter


class SalesManager:

    accommodationList = []

    def find_accommodation_by_price(price, accommodation_list):
        found_accommodation_by_price = []
        for accommodation in accommodation_list:
            if accommodation.price < price:
                found_accommodation_by_price.append(accommodation)
        return found_accommodation_by_price

    def find_by_distance(accommodation_list, location_of_sio, distance):
        found_accommodation_by_distance = []
        for accommodation in accommodation_list:
            if (accommodation.latitude - location_of_sio) < distance:
                found_accommodation_by_distance.append(accommodation)

        return found_accommodation_by_distance

    def sort_accommodation_by_price(accommodation_list):
        return sorted(accommodation_list, key=attrgetter('price'))